### PR TITLE
fix: refacto context

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -300,33 +300,40 @@ A TestCase Context has to implement this interface
...
 // New returns a new TestCaseContext
 func New() venom.TestCaseContext {
-	return &TestCaseContext{}
+	ctx := &TestCaseContext{}
+	ctx.Name = Name
+	return ctx
 }
 
-// TestCaseContex represents the context of a testcase
+// TestCaseContext represents the context of a testcase
```
Comment (yesnault) [inline README.md]: Can you rename this struct ?  it's little confusing with TestCaseContext Interface

File: README.md [hunk 2]
```diff
@@ -300,33 +300,40 @@ A TestCase Context has to implement this interface
...
+func (tcc *TestCaseContext) Init() error {
+	return nil
+}
 
-	vars[ContextFooKey] = 'fooValue'
-	vars[ContextBarKey] = 'varValue'
-	return vars, nil
+// Close the context
+func (tcc *TestCaseContext) Close() error {
+	return nil
```
Comment (yesnault) [inline README.md]: Others func from interface TestCaseContext are missing here

File: context/default/context.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+// New returns a new TestCaseContext
+func New() venom.TestCaseContext {
+	ctx := &TestCaseContext{}
+	ctx.Name = Name
+	return ctx
+}
+
+// TestCaseContext represents the context of a testcase
+type TestCaseContext struct {
```
Comment (yesnault) [inline context/default/context.go]: Can you rename on DefaultTestCaseContext ? (to avoid confusion with Interface name)

File: venom.go [hunk 4]
```diff
@@ -29,16 +28,17 @@ func RegisterExecutor(name string, e Executor) {
...
+func getExecutorWrap(t map[string]interface{}, tcc TestCaseContext) (*executorWrap, error) {
 	var name string
 	var retry, delay, timeout int
 
 	if itype, ok := t["type"]; ok {
 		name = fmt.Sprintf("%s", itype)
 	}
 
-	if name == "" {
+	if name == "" && tcc.GetName() != "default" {
```
Comment (yesnault) [inline venom.go]: This should be in documentation somewhere, no?

**Comment by yesnault** [comment]
A common use case (Testcase):
- Go to an URL, click registrer -> a mail is send by application tested
- Go on mailbox (new executor IMAP is coming soon) -> check mail, get link to go
- Go to link to confirm inscription

-> This use case should be possible, so with 2 context (web & imap). It is possible to add many context on a TestCase ?

File: README.md (hunk 1 above)
Comment (sguiheux) [inline README.md]: In readme only?

Comment (yesnault) [inline README.md]: yep

**Pull request merged**